packages/kit/src/exports/vite/index.js

```diff
@@ -333,17 +261,152 @@ function kit({ svelte_config }) {
+
+				// add entry points for every endpoint...
+				manifest_data.routes.forEach((route) => {
+					if (route.endpoint) {
+						const resolved = path.resolve(route.endpoint.file);
+						const relative = decodeURIComponent(path.relative(kit.files.routes, resolved));
+						const name = posixify(path.join('entries/endpoints', relative.replace(/\.js$/, '')));
+						input[name] = resolved;
+					}
+				});
+
+				// ...and every component used by pages...
+				manifest_data.nodes.forEach((node) => {
+					for (const file of [node.component, node.universal, node.server]) {
+						if (file) {
+							const resolved = path.resolve(file);
+							const relative = decodeURIComponent(path.relative(kit.files.routes, resolved));
+
+							const name = relative.startsWith('..')
+								? posixify(path.join('entries/fallbacks', path.basename(file)))
+								: posixify(path.join('entries/pages', relative.replace(/\.js$/, '')));
+							input[name] = resolved;
+						}
+					}
+				});
+
+				// ...and every matcher
+				Object.entries(manifest_data.matchers).forEach(([key, file]) => {
+					const name = posixify(path.join('entries/matchers', key));
+					input[name] = path.resolve(file);
+				});
+			} else {
+				/** @type {Record<string, string>} */
+				input.start = `${runtime_directory}/client/start.js`;
+
+				manifest_data.nodes.forEach((node) => {
+					if (node.component) {
+						const resolved = path.resolve(node.component);
+						const relative = decodeURIComponent(
+							path.relative(svelte_config.kit.files.routes, resolved)
+						);
+
+						const name = relative.startsWith('..')
+							? path.basename(node.component)
+							: posixify(path.join('pages', relative));
+						input[`components/${name}`] = resolved;
+					}
+
+					if (node.universal) {
+						const resolved = path.resolve(node.universal);
+						const relative = decodeURIComponent(
+							path.relative(svelte_config.kit.files.routes, resolved)
+						);
+
+						const name = relative.startsWith('..')
+							? path.basename(node.universal)
+							: posixify(path.join('pages', relative));
+						input[`modules/${name}`] = resolved;
+					}
+				});
+			}
+
+			/** @type {import('vite').UserConfig} */
+			const new_config = {
+				base: ssr ? assets_base(kit) : './',
+				build: {
+					cssCodeSplit: true,
+					outDir: `${out}/${ssr ? 'server' : 'client'}`,
+					rollupOptions: {
+						input,
+						output: {
+							format: 'esm',
+							entryFileNames: ssr ? '[name].js' : `${prefix}/[name]-[hash].js`,
+							chunkFileNames: ssr ? 'chunks/[name].js' : `${prefix}/chunks/[name]-[hash].js`,
+							assetFileNames: `${prefix}/assets/[name]-[hash][extname]`,
+							hoistTransitiveImports: false
+						},
+						preserveEntrySignatures: 'strict'
+					},
+					target: ssr ? 'node16.14' : undefined,
+					// don't use the default name to avoid collisions with 'static/manifest.json'
+					manifest: 'vite-manifest.json'
+				},
+				define: {
```

Comment:
this kind of stuff seems like it belongs in the setup plugin

Reply:
why? the `define` values for build are different than for dev

Reply:
I intended the setup plugin to contain all the setup regardless of whether it's dev or build - it can return different config depending on which command is run. Tools like Storybook will still need things like `__SVELTEKIT_DEV__` set appropriately in build mode.

Reply:
Do they? I think that stuff's only use by SvelteKit internals that shouldn't be imported in a Storybook context. Ww _can_ put it in `setup` but it'll increase complexity

Reply:
Users may create components that reference [SvelteKit modules](https://kit.svelte.dev/docs/modules). If those modules use defines that are not available or set properly it will end up causing issues.